packages/@aws-cdk/aws-codepipeline-actions/test/stepfunctions/test.stepfunctions-invoke-actions.ts

```diff
@@ -0,0 +1,175 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as stepfunction from '@aws-cdk/aws-stepfunctions';
+import { Stack } from '@aws-cdk/core';
+import { Test } from 'nodeunit';
+import * as cpactions from '../../lib';
+import { StateMachineInputType } from '../../lib';
+
+export = {
+  'StepFunctions Invoke Action': {
+    'Verify stepfunction configuration properties are set to specific values'(test: Test) {
+      const stack = new Stack();
+
+      //when
+      minimalPipeline(stack, undefined);
+
+      //then
+      expect(stack).to(haveResourceLike('AWS::CodePipeline::Pipeline', {
+        'Stages': [
+          //Must have a source stage
+          {
+            'Actions': [
+              {
+                ActionTypeId: {
+                  Category: 'Source',
+                  Owner: 'AWS',
+                  Provider: 'S3',
+                  Version: '1',
+                },
+                'Configuration': {
+                  S3Bucket: {
+                    Ref: 'MyBucketF68F3FF0',
+                  },
+                  S3ObjectKey: 'some/path/to',
+                },
+              },
+              
+            ],
+          },
+          // Must have stepfunction invoke action configuration
+          {
+            'Actions': [
+              {
+                ActionTypeId: {
+                  Category: 'Invoke',
+                  Owner: 'AWS',
+                  Provider: 'StepFunctions',
+                  Version: '1',
+                },
+                'Configuration': {
+                  StateMachineArn: {
+                    Ref: 'SimpleStateMachineE8E2CF40',
+                  },
+                  InputType: 'Literal',
+                },
+              },
+            ],
+          },
+        ],
+      }));
+
+      expect(stack).to(not(haveResourceLike('AWS::Events::Rule')));
+
+      test.done();
+    }, 
+
+    'Allows the pipeline to invoke this stepfunction'(test: Test) {
+      const stack = new Stack();
+
+      minimalPipeline(stack, undefined);
+      
+      expect(stack).to(haveResourceLike('AWS::IAM::Policy', {
+        'PolicyDocument': {
+          'Statement': [
+            {
+              'Action': ['states:StartExecution','states:DescribeStateMachine'],
+              'Resource': {
+                'Ref': 'SimpleStateMachineE8E2CF40',
+              },
+              'Effect': 'Allow',
+            },
+          ],
+        },
+      }));
+
+      expect(stack).to(haveResourceLike('AWS::IAM::Role'));
+
+      test.done();
+    }, 
+
+    'does not allow passing empty StateMachineInput when the StateMachineInputType is FilePath'(test: Test) {
+      const stack = new Stack();
+
+      const startState = new stepfunction.Pass(stack, 'StartState');
+
+      const simpleStateMachine  = new stepfunction.StateMachine(stack, 'SimpleStateMachine', {
+        definition: startState,
+      });
+
+      test.throws(() => {
```

Comment:
I think you should test what it throws, exactly. So you know it's not like UndefinedException or whatever